content/learning-paths/servers-and-cloud-computing/cca-essentials/example.md

```diff
@@ -0,0 +1,174 @@
+
+```output
+Starting syslogd: OK
+Starting klogd: OK
+Running sysctl: OK
+Starting network: udhcpc: started, v1.36.1
+udhcpc: broadcasting discover
+udhcpc: broadcasting select for 192.168.33.15, server 192.168.33.1
+udhcpc: lease of 192.168.33.15 obtained from 192.168.33.1, lease time 14400
+deleting routers
+adding dns 172.20.51.254
+OK
+
+Welcome to the CCA realm
+realm login: root
+(realm) #
+```
+
+Now run the key broker client application in the realm. Use the endpoint address that the key broker server is listening on in the other terminal:
+```bash
+cd /cca 
+./keybroker-app -v --endpoint http://172.17.0.2:8088 skywalker 
+```
+In the command above `skywalker` is the key name that is requested from the key broker server. After some time, you should see the following output:
+```
+INFO Requesting key named 'skywalker' from the keybroker server with URL http://172.17.0.2:8088/keys/v1/key/skywalker
+INFO Challenge (64 bytes) = [0f, ea, c4, e2, 24, 4e, fa, dc, 1d, ea, ea, 3d, 60, eb, a6, 8f, f1, ed, 1a, 07, 35, cb, 5b, 1b, cf, 5b, 21, a4, bc, 14, 65, c2, 21, 3f, bf, 33, a0, b0, 7c, 78, 3a, a6, 32, c6, 34, be, ff, 45, 98, f4, 17, b1, 24, 71, 4f, 9c, 75, 58, 37, 3a, 28, ea, 97, 33]
+INFO Submitting evidence to URL http://172.17.0.2:8088/keys/v1/evidence/3974368321
+INFO Attestation failure :-( ! AttestationFailure: No attestation result was obtained. No known-good reference values.
+```
+You can see from the key broker client application output that the `skywalker` key is requested from the key broker server, which did send a challenge. The key broker client application uses the challenge to submit its evidence back to the key broker server, but it gets an attestation failure. This is because the server does not have any known good reference values.
+
+Now look at the key broker server output on the terminal where the server is running. It will look like:
+
+```output
+INFO Known-good RIM values are missing. If you trust the client that submitted
+evidence for challenge 1302147796, you should restart the keybroker-server with the following
+command-line option to populate it with known-good RIM values:
+--reference-values <(echo '{ "reference-values": [ "tiA66VOokO071FfsCHr7es02vUbtVH5FpLLqTzT7jps=" ] }')
+INFO Evidence submitted for challenge 1302147796: no attestation result was obtained. No known-good reference values.
+```
+From the server output you will notice that it did create the challenge for the key broker application, but it complains that it has no known good reference values. It does however provide a way to provision the key broker server with known good values if the client is trusted. 
+In a production environment, the known good reference value would be generated, but for demonstration purposes and simplification, you will use the value proposed by the key broker server.
+
+Now go ahead and terminate the running instance of the key broker server(ctrl+C) and restart it with the known good reference value:
```

Comment:
I think we need to emphasise that we are copy/pasting the `--reference-values` parameter from the error message that was generated on the previous run. The golden value in the following command is _probably_ going to be correct given that we are using fixed versions of everything, but there is some risk of the user running the command verbatim, and not actually picking up the golden value that the KBS has indicated. If they do this, the attestation will fail again. Perhaps just add an extra note along the following lines:

Notice here that you need to copy the `--reference-values` argument directly from the previous error message reported by the key broker. When running this next command, ensure that you are using exactly that value, for example:

Reply:
addressed.